changelog.d/10581.bugfix

```diff
@@ -0,0 +1 @@
+Remove pushers when deleting 3pid from account.
```

Comment:
Additionally, could you add an entry to [CHANGES.md](https://github.com/matrix-org/synapse/blob/develop/CHANGES.md) and [docs/upgrade.md](https://github.com/matrix-org/synapse/blob/develop/docs/upgrade.md) just informing homeserver operators that users may stop receiving message updates via email for addresses that were once, but not longer linked to their account?

Just sticking it at the top of CHANGES.md should be fine, we'll just need to remember to hoist it up to the right section come release day. For docs/upgrade.md, simply make a new section for the next release, likely 1.42.0, and put the notice in there under an appropriate heading.